internal/telemetry/tracker.go

```diff
@@ -26,15 +34,28 @@ func (tracker *stdoutTracker) Track(event event) {
 	)
 }
 
-type segmentTracker struct{}
+type segmentTracker struct {
+	client analytics.Client
+	logger *log.Logger
+}
+
+func newSegmentTracker(logger *log.Logger) Tracker {
+	if len(segmentWriteKey) == 0 {
+		log.Print("unable to make a Segment tracker with an empty write key; replacing with a noop")
+		return &noopTracker{}
+	}
+	client := analytics.New(segmentWriteKey)
+	return &segmentTracker{client, logger}
+}
 
-// TODO(REALMC-7243): use Segment sdk to send events through client
 func (tracker *segmentTracker) Track(event event) {
-	fmt.Printf(
-		"%s UTC TELEM %s: %s%v\n",
-		event.time.In(time.UTC).Format("15:04:05"),
-		event.command,
-		event.eventType,
-		event.data,
-	)
+	if err := tracker.client.Enqueue(analytics.Track{
+		MessageId:  primitive.NewObjectID().Hex(),
+		Timestamp:  time.Now(),
+		Event:      string(event.eventType),
+		UserId:     event.userID,
+		Properties: eventDataProperties(event.data),
```

Comment:
I might have sent you down a bad path here, but we probably don't even need `eventDataProperties` as a function (and could/should handle that logic just here in this function).

the thing that's missing right now are two additional `EventData`s we want to care about:

```go
[]EventData{
  {"cmd", event.command}, // "cmd" => eventDataKeyCommand
  {"xid", event.executionID}, // "xid" => eventDataKeyExecutionID
}
```

since this is all just segment-specific stuff, i think using `make(map[string]interface{}, len(event.data)+2)` inside this function before the `client.Enqueue` call works fine here